### ISSUE TITLE
Update Websocket++ to 0.6.0

### ISSUE THREAD

**Issue by ras0219-msft** [Microsoft/cpprestsdk#33]
Crossposted from: https://casablanca.codeplex.com/discussions/648800

> Several months ago, version 0.6.0 of WebSocket++ was released: 
> https://github.com/zaphoyd/websocketpp/releases 
> It brings some compelling features like Boost-less ASIO, which eliminates the need for Boost binaries for iOS and Android. But what interest me most is the upcoming support for permessage-deflate (linked from the above page,  https://github.com/zaphoyd/websocketpp/archive/0.6.0-permessage-deflate.zip), which has the potential of compressing WebSocket data a lot. 
> 
> Are their any plans to update WebSocket++ in Casablanca to the latest versions? 
> 
> Thanks, 
> Boris

**Comment by ras0219-msft** [comment]
This would be great to do. Any PRs for this would be gladly accepted :+1:.

Bonus points if we can remove the Boost dependency on Windows.

**Comment by coliemcgarry** [comment]
Hi there,
I build cpprest sdk with the WebSocketpp 0.6.0, and all seems well (Windows platform). I did need to disable some warnings by defining _SCL_SECURE_NO_WARNINGS (WebSocketpp is using some std::copy() calls which cause warning, and because precompiled headers are used in cpprest, I needed to disable at a project level i.e. as opposed to disabling at a file level.).

Anyway, regarding removing the boost dependency: From what I can determine boost is still required by Cert Utilities (x509_cert_utilities.cpp). So I guess boost needs to remain a dependency for windows just now. 

I can post a pull request for the changes I made to build with WebSocketpp, but there's very little change!

Thanks,
Colie
